### PR TITLE
Feature/573 characteristics filter

### PR DESCRIPTION
**Pull request by maxdiebold-erg**
## Related Issues:
* [HMW-573](https://jira.epa.gov/browse/HMW-573)

## Main Changes:
* Added a multi-selectable characteristic filter to the _Past Water Conditions_ accordion lists. This input's options are drawn from the WQP `periodOfRecord` summary service, so is not immediately active.
* Moved the point of fetching `periodOfRecord` data up into the `useMonitoringLocationsLayers` hook. There is an option to disable, which is used on the Monitoring Report page. While the service may still be used on that page in [HMW-572](https://jira.epa.gov/browse/HMW-572), it will likely be on an as-needed basis.
* Updated some state items to use sensible defaults instead of `null` (much more feasible now due to the amount of TypeScript).

## Steps To Test:
1. Navigate to http://localhost:3000/community/dc/monitoring.
2. In the _Past Water Conditions_ tab, select a characteristic from the **Filter by Characteristic** input, and confirm the change is reflected in the list.
3. Go to the Monitoring Report page of several of those list items and confirm the selected characteristic is indeed present.
4. On the Community page, ensure that the date slider continues to filter locations alongside the characteristic filter.
5. Go to the _Water Monitoring Locations_ section of the Overview tab.
6. Confirm the characteristic filter is present.
7. Toggle of the Past Water Conditions, then confirm the characteristic filter is no longer visible.
8. Go to http://localhost:3000/community/Portland,%20OR,%20USA/monitoring, and verify the toggle table and list items are still populated before the date slider and characteristic input are ready.
9. Confirm the filter works properly for the list view on tribal pages, e.g. http://localhost:3000/tribe/UTEMTN.

### CODE REVIEW

**Comment by maxdiebold-erg** [comment]
I introduced a bug into the `MonitoringLocationsContent`, since the `timeframe` property is only calculated on the Water Monitoring tab. I think we could go back to an initial `null` value for that property, or set it to each station's measurement date range.

**Comment by maxdiebold-erg** [comment]
I reverted that behavior for now. If we want to show each station's timeframe on other tabs, then the behavior on the Monitoring tab should probably be adjusted: instead of showing the currently selected range in each location's panel, it would make more sense to show the station's measurement timeframe _bounded_ by the selected range (i.e. the max of the station's min and the date slider min, up to the min of the station's max and the date slider max).

**Review by cschwinderg — CHANGES_REQUESTED** [review]
Works well! 

I think the characteristic filter should also filter the points on the map like the filters above (i.e., Bacterial, Metals, etc.). The number of items on the map should match the number of items in the list.

File: app/client/src/components/shared/TribalMapList.js [hunk 1]
```diff
@@ -958,20 +959,50 @@ function MonitoringTab({ activeState }: MonitoringTabProps) {
...
+
   const [expandedRows, setExpandedRows] = useState([]);
 
   return (
     <AccordionList
       title={
         <>
-          <strong>{sortedMonitoringAndSensors.length}</strong> locations with
-          data in the <em>{activeState.label}</em> watershed.
+          <strong>{filteredMonitoringAndSensors.length}</strong> locations with{' '}
```
Comment (cschwinderg) [inline app/client/src/components/shared/TribalMapList.js:986]:
```suggestion
          <strong>{filteredMonitoringAndSensors.length}</strong> of <strong>{sortedMonitoringAndSensors.length}</strong>  {sortedMonitoringAndSensors.length === 1 ? 'location' : 'locations'} with{' '}
```

This should include text like what we have on the community monitoring tab (i.e., `X of Y locations with...`).

**Review by cschwinderg — CHANGES_REQUESTED** [review]

**Comment by maxdiebold-erg** [comment]
@cschwinderg  Good points! I'll make those changes.

File: app/client/src/components/pages/Community.Tabs.Monitoring.js [hunk 2]
```diff
@@ -1319,23 +1263,33 @@ function PastConditionsTab({ setMonitoringDisplayed }) {
...
-                    of <strong>{totalLocationsCount.toLocaleString()}</strong>{' '}
-                    water monitoring sample locations in the{' '}
-                    <em>{watershed}</em> watershed.
-                  </span>
-                )
+                <span data-testid="monitoring-accordion-title">
+                  <strong>{displayedLocationsCount.toLocaleString()}</strong> of{' '}
+                  <strong>{totalLocationsCount.toLocaleString()}</strong> water
+                  monitoring sample{' '}
+                  {totalLocationsCount === 1 ? 'location' : 'locations'}
```
Comment (cschwinderg) [inline app/client/src/components/pages/Community.Tabs.Monitoring.js:1270]:
```suggestion
                  monitoring sample locations
```

This was totally my fault. I didn't check the code I copied from close enough. Using the singular for location here is gramatically incorrect. It is also incorrect in the location that I copied from as well.

Current:
`1 of 1 water monitor sample location with the selected characteristic in the "some place" watershed from 1999 to 2023.`

Suggestion:
`1 of 1 water monitor sample locations with the selected characteristic in the "some place" watershed from 1999 to 2023.`

I can go ahead and commit this change.

Comment (maxdiebold-erg) [reply app/client/src/components/pages/Community.Tabs.Monitoring.js]: I thought that sounded off... thanks for making the fixes.

**Review by cschwinderg — APPROVED** [review]
Works well! Thank you for making those changes.

**Pull request merged**